### PR TITLE
MODULES-5428 : Add passenger setting support inside Directory directive

### PR DESCRIPTION
**Pull request by dacat**
I added support for the passenger settings you can use inside Directory as they are documented at https://www.phusionpassenger.com/library/config/apache/reference/. 

* This branched off of master and not the other passenger feature branch I created.
* The README.md changes are more detailed in this pull request as it also details which passenger options are supported where. ( It may not be appropriate to to have the 'context' column inside the documentation for the mod::passenger class.

### CODE REVIEW

File: spec/acceptance/mod_passenger_spec.rb [hunk 1]
```diff
@@ -86,123 +171,175 @@
...
+          EOS
+         apply_manifest(pp, :catch_failures => true)
+       end
+       describe file("#{$vhost_dir}/25-passenger.example.com.conf") do
+         all_passenger_directory_options = passenger_config_options.select {|k,v| /directory/ =~ v[:Context]}
+         all_passenger_directory_options.each do |k,v|
+           case v[:Type]
+             when 'QuotedString', 'RelPath', 'FullPath', 'Path'
+              it { is_expected.to contain "#{v[:PassOpt]} \"something\"" }
+             when 'String', 'URI', 'Integer'
```
Comment (HAIL9000) [inline spec/acceptance/mod_passenger_spec.rb]: It's not clear to me why `String`, `URI`,  and `Integer` need their own case? It looks like the code is functionally the same to the else statement below.

**Comment by HAIL9000** [comment]
Just making a note, it looks like this PR features the same update to the README as #1665 so once we merge one PR we'll want to rebase the other.

**Comment by dacat** [comment]
@HAIL9000 I made a comment in the other PR about having a new branch with all changes in one. It might be preferred if I finish up all of it then re-request.  Let me know what you think.

**Comment by HAIL9000** [comment]
Hey @dacat, we just merged the other passenger PR #1665. Now that it's in, would you mind rebasing this one so we can merge it cleanly?

Thank you!

**Comment by dacat** [comment]
@HAIL9000 I took a look and its going to take a bit for me to fix the conflict. Just wanted to drop a note to say I'm working on it.

**Comment by HAIL9000** [comment]
@dacat Thanks for the update. And no worries! We'll review it whenever you're ready, thanks for all your hard work

**Comment by david22swan** [comment]
@dacat Was wondering how close this is to ready, a clean release for Rubocopification is planned  and I would prefer that it get's in before then to prevent you from having to rebase?

**Comment by david22swan** [comment]
Closing and reopening PR to solve bundler issue

**Comment by david22swan** [comment]
@dacat Apologies but due to the length of time this has been left fallow and the conflicts that exit between files I am choosing to close this PR. I apologize if this inconveniences you and if you wish to reopen it in the future, we at the Modules team will be ore than happy to review it.
Best Wishes

**Pull request closed without merging**